pandas/core/indexes/datetimes.py

```diff
@@ -489,6 +489,9 @@ def union(self, other, sort=None):
                 other = DatetimeIndex(other)
             except TypeError:
                 pass
+            except ValueError:
+                # GH 26283: Convert indexes with mixed tz to UTC
+                other = tools.to_datetime(other, utc=True)
```

Comment:
this is not correct; rather you should call the super.union() as these must be combined as Index rather than DTI

Reply:
So `super().union` actually returns the same result because these evaluate as equal:

```
>>> Timestamp('2019-05-01 00:00:00', tz='UTC') == Timestamp('2019-05-01 01:00:00', tz='Europe/London')
True
```

While an Index with all Timestamps feels a little more correct, a DTI with UTC timestamps is technically correct too since the above is `True`. (And less code to handle this special casing is a nice plus)

Reply:
I disagree, we are striving *never* to coerce to UTC unless its explicit, this should stay as ``object`` dtyped.